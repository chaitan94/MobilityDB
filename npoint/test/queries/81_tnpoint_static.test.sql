/******************************************************************************
 * Input
 ******************************************************************************/

SELECT npoint 'npoint(1,0.5)';
SELECT npoint ' npoint   (   1   ,	0.5   )   ';

SELECT nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment '  nsegment  (  1  ,  0.5  ,  0.7 ) ';

/******************************************************************************
 * Constructors
 ******************************************************************************/

SELECT npoint(1, 0.5);

SELECT nsegment(1, 0.2, 0.6);
SELECT nsegment(1);
SELECT nsegment(1, 0.2);
SELECT nsegment(npoint(1, 0.5));

/******************************************************************************
 * Accessing values
 ******************************************************************************/

SELECT route(npoint 'npoint(1,0.5)');
SELECT getPosition(npoint 'npoint(1,0.5)');

SELECT route(nsegment 'nsegment(1,0.5,0.7)');
SELECT startPosition(nsegment 'nsegment(1,0.5,0.7)');
SELECT endPosition(nsegment 'nsegment(1,0.5,0.7)');

/******************************************************************************
 * Cast functions between network and space
 ******************************************************************************/

SELECT st_astext(npoint 'npoint(1,0.2)'::geometry);

SELECT st_astext(nsegment 'nsegment(1,0.5,0.7)'::geometry);

SELECT (npoint 'npoint(1,0.2)'::geometry)::npoint;

SELECT (nsegment 'nsegment(1,0.5,0.7)'::geometry)::nsegment;

SELECT geometry 'Point(610.455019399524 528.508247341961)'::npoint;

SELECT geometry 'LINESTRING(416.346567736997 528.335344322874,610.455019399524 528.508247341961,476.989195102204 642.550969672973)'::nsegment;

/******************************************************************************
 * Comparisons
 ******************************************************************************/

SELECT npoint 'npoint(1,0.5)' = npoint 'npoint(1,0.5)';
SELECT npoint 'npoint(1,0.5)' = npoint 'npoint(1,0.7)';
SELECT npoint 'npoint(1,0.5)' = npoint 'npoint(2,0.5)';

SELECT npoint 'npoint(1,0.5)' != npoint 'npoint(1,0.5)';
SELECT npoint 'npoint(1,0.5)' != npoint 'npoint(1,0.7)';
SELECT npoint 'npoint(1,0.5)' != npoint 'npoint(2,0.5)';

SELECT npoint 'npoint(1,0.5)' < npoint 'npoint(1,0.5)';
SELECT npoint 'npoint(1,0.5)' < npoint 'npoint(1,0.7)';
SELECT npoint 'npoint(1,0.5)' < npoint 'npoint(2,0.5)';

SELECT npoint 'npoint(1,0.5)' <= npoint 'npoint(1,0.5)';
SELECT npoint 'npoint(1,0.5)' <= npoint 'npoint(1,0.7)';
SELECT npoint 'npoint(1,0.5)' <= npoint 'npoint(2,0.5)';

SELECT npoint 'npoint(1,0.5)' > npoint 'npoint(1,0.5)';
SELECT npoint 'npoint(1,0.5)' > npoint 'npoint(1,0.7)';
SELECT npoint 'npoint(1,0.5)' > npoint 'npoint(2,0.5)';

SELECT npoint 'npoint(1,0.5)' >= npoint 'npoint(1,0.5)';
SELECT npoint 'npoint(1,0.5)' >= npoint 'npoint(1,0.7)';
SELECT npoint 'npoint(1,0.5)' >= npoint 'npoint(2,0.5)';

SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(1,0.3,0.5)';
SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(1,0.3,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(2,0.3,0.5)';

SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(1,0.3,0.5)';
SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(1,0.3,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(2,0.3,0.5)';

SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(1,0.3,0.5)';
SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(1,0.3,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(2,0.3,0.5)';

SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(1,0.3,0.5)';
SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(1,0.3,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(2,0.3,0.5)';

SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(1,0.3,0.5)';
SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(1,0.3,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(2,0.3,0.5)';

SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(1,0.3,0.5)';
SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(1,0.3,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(1,0.5,0.7)';
SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(2,0.3,0.5)';

/******************************************************************************/
