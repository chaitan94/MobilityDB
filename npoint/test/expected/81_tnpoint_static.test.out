SELECT npoint 'npoint(1,0.5)';
    npoint     
---------------
 NPoint(1,0.5)
(1 row)

SELECT npoint ' npoint   (   1   ,	0.5   )   ';
    npoint     
---------------
 NPoint(1,0.5)
(1 row)

/* Errors */
SELECT npoint 'point(1,0.5)';
ERROR:  Could not parse network point
LINE 2: SELECT npoint 'point(1,0.5)';
                      ^
SELECT npoint 'npoint(1,0.5';
ERROR:  Could not parse network point
LINE 1: SELECT npoint 'npoint(1,0.5';
                      ^
SELECT npoint 'npoint(1 0.5)';
ERROR:  Could not parse network point
LINE 1: SELECT npoint 'npoint(1 0.5)';
                      ^
SELECT npoint 'npoint(1000,0.5)';
ERROR:  there is no route with gid value 1000 in table ways
LINE 1: SELECT npoint 'npoint(1000,0.5)';
                      ^
SELECT npoint 'npoint(1,1.5)';
ERROR:  The relative position must be a real number between 0 and 1
LINE 1: SELECT npoint 'npoint(1,1.5)';
                      ^
SELECT nsegment 'nsegment(1,0.5,0.7)';
      nsegment       
---------------------
 NSegment(1,0.5,0.7)
(1 row)

SELECT nsegment '  nsegment  (  1  ,  0.5  ,  0.7 ) ';
      nsegment       
---------------------
 NSegment(1,0.5,0.7)
(1 row)

/* Errors */
SELECT nsegment 'segment(1,0.5,0.7)';
ERROR:  Could not parse network segment
LINE 2: SELECT nsegment 'segment(1,0.5,0.7)';
                        ^
SELECT nsegment 'nsegment(1,0.5,0.7';
ERROR:  Could not parse network segment
LINE 1: SELECT nsegment 'nsegment(1,0.5,0.7';
                        ^
SELECT nsegment 'nsegment(1 0.5 0.7)';
ERROR:  Could not parse network segment
LINE 1: SELECT nsegment 'nsegment(1 0.5 0.7)';
                        ^
SELECT nsegment 'nsegment(1000,0.5,0.7)';
ERROR:  there is no route with gid value 1000 in table ways
LINE 1: SELECT nsegment 'nsegment(1000,0.5,0.7)';
                        ^
SELECT nsegment 'nsegment(1,1.5,0.7)';
ERROR:  the relative position must be a real number between 0 and 1
LINE 1: SELECT nsegment 'nsegment(1,1.5,0.7)';
                        ^
SELECT npoint(1, 0.5);
    npoint     
---------------
 NPoint(1,0.5)
(1 row)

/* Errors */
SELECT npoint(1000,0.5);
ERROR:  there is no route with gid value 1000 in table ways
SELECT npoint(1,1.5);
ERROR:  the relative position must be a real number between 0 and 1
SELECT nsegment(1, 0.2, 0.6);
      nsegment       
---------------------
 NSegment(1,0.2,0.6)
(1 row)

SELECT nsegment(1);
    nsegment     
-----------------
 NSegment(1,0,1)
(1 row)

SELECT nsegment(1, 0.2);
     nsegment      
-------------------
 NSegment(1,0.2,1)
(1 row)

SELECT nsegment(npoint(1, 0.5));
      nsegment       
---------------------
 NSegment(1,0.5,0.5)
(1 row)

/* Errors */
SELECT nsegment(1000,0.5,0.7);
ERROR:  there is no route with gid value 1000 in table ways
SELECT nsegment(1,1.5,0.7);
ERROR:  The relative position of a network segment must be a real number between 0 and 1
SELECT route(npoint 'npoint(1,0.5)');
 route 
-------
     1
(1 row)

SELECT getPosition(npoint 'npoint(1,0.5)');
 getposition 
-------------
         0.5
(1 row)

SELECT route(nsegment 'nsegment(1,0.5,0.7)');
 route 
-------
     1
(1 row)

SELECT startPosition(nsegment 'nsegment(1,0.5,0.7)');
 startposition 
---------------
           0.5
(1 row)

SELECT endPosition(nsegment 'nsegment(1,0.5,0.7)');
 endposition 
-------------
         0.7
(1 row)

SELECT st_astext(npoint 'npoint(1,0.2)'::geometry);
                st_astext                
-----------------------------------------
 POINT(69.790368869663 81.4520975905736)
(1 row)

SELECT st_astext(nsegment 'nsegment(1,0.5,0.7)'::geometry);
                                    st_astext                                    
---------------------------------------------------------------------------------
 LINESTRING(48.7186629128278 77.7640705101509,34.6506927416814 75.3845848205875)
(1 row)

SELECT (npoint 'npoint(1,0.2)'::geometry)::npoint;
    npoint     
---------------
 NPoint(1,0.2)
(1 row)

SELECT (nsegment 'nsegment(1,0.5,0.7)'::geometry)::nsegment;
      nsegment       
---------------------
 NSegment(1,0.5,0.7)
(1 row)

SELECT geometry 'Point(610.455019399524 528.508247341961)'::npoint;
 npoint 
--------
 
(1 row)

SELECT geometry 'LINESTRING(416.346567736997 528.335344322874,610.455019399524 528.508247341961,476.989195102204 642.550969672973)'::nsegment;
 nsegment 
----------
 
(1 row)

SELECT npoint 'npoint(1,0.5)' = npoint 'npoint(1,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' = npoint 'npoint(1,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' = npoint 'npoint(2,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' != npoint 'npoint(1,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' != npoint 'npoint(1,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' != npoint 'npoint(2,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' < npoint 'npoint(1,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' < npoint 'npoint(1,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' < npoint 'npoint(2,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' <= npoint 'npoint(1,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' <= npoint 'npoint(1,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' <= npoint 'npoint(2,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' > npoint 'npoint(1,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' > npoint 'npoint(1,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' > npoint 'npoint(2,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' >= npoint 'npoint(1,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT npoint 'npoint(1,0.5)' >= npoint 'npoint(1,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT npoint 'npoint(1,0.5)' >= npoint 'npoint(2,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(1,0.3,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(1,0.3,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(1,0.5,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' = nsegment 'nsegment(2,0.3,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(1,0.3,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(1,0.3,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(1,0.5,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' != nsegment 'nsegment(2,0.3,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(1,0.3,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(1,0.3,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(1,0.5,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' < nsegment 'nsegment(2,0.3,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(1,0.3,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(1,0.3,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(1,0.5,0.7)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' <= nsegment 'nsegment(2,0.3,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(1,0.3,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(1,0.3,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(1,0.5,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' > nsegment 'nsegment(2,0.3,0.5)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(1,0.3,0.5)';
 ?column? 
----------
 t
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(1,0.3,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(1,0.5,0.7)';
 ?column? 
----------
 f
(1 row)

SELECT nsegment 'nsegment(1,0.3,0.5)' >= nsegment 'nsegment(2,0.3,0.5)';
 ?column? 
----------
 f
(1 row)

